src/storage.cc

```diff
@@ -158,8 +158,15 @@ void Storage::Tock() {
   std::vector<double>::iterator result;
   result = std::max_element(in_commod_prefs.begin(), in_commod_prefs.end());
   int maxindx = std::distance(in_commod_prefs.begin(), result);
-  cyclus::toolkit::RecordTimeSeries<double>("demand"+in_commods[maxindx], this,
+  
+  if (manager()->context()->time() % (active_buying + dormant_buying) < active_buying) {
+    cyclus::toolkit::RecordTimeSeries<double>("demand"+in_commods[maxindx], this,
                                             current_capacity());
+  }
+  else {
+    cyclus::toolkit::RecordTimeSeries<double>("demand"+in_commods[maxindx], this, 0);
+    }
+  
```

Comment:
This is a little less cognitive burden and less cut & paste code.
```suggestion
  double demand = 0;
  if (manager()->context()->time() % (active_buying + dormant_buying) < active_buying) {
    demand = current_capacity());
  }
  
  cyclus::toolkit::RecordTimeSeries<double>("demand"+in_commods[maxindx], this, demand);
  
```